velodyne_driver/src/driver/driver.cc

```diff
@@ -237,6 +264,14 @@ bool VelodyneDriver::poll(void)
       	 last_azimuth_ = azimuth;
       	 continue;
       }
+
+      if((last_azimuth_ < config_.timestamp_angle && config_.timestamp_angle <= azimuth)
+         || ( config_.timestamp_angle <= azimuth && azimuth < last_azimuth_)
+         || (azimuth < last_azimuth_ && last_azimuth_ < config_.timestamp_angle))
```

Comment:
The logic in this if statement is the same logic as line 275 and should really be placed in its own function to be used on both lines
```
if (angleIsBetween(last_azimuth_, azimuth, config_.timestamp_angle))
```

Within the function, I think you need a comment just explaining the logic since it was not immediately obvious to me.
```
// Case 1: 0° < last azimuth < angle < azimuth < 360°
// Case 2: last azimuth < 360° < 0° < angle < azimuth
// Case 3: last azimuth < angle < 360° < 0° < azimuth
```

I'm not even completely sure what is going on since there are other cases listed in the issue here but I don't know how those show up https://github.com/ros-drivers/velodyne/issues/174